bullet_train-api/app/controllers/concerns/api/controllers/base.rb

```diff
@@ -30,18 +27,24 @@ def modify_url_params(url, new_params)
 
     def set_pagination_headers
       return unless @pagy
-
-      if @pagy.has_more?
-        if (collection = instance_variable_get(collection_variable))
-          next_cursor = collection.last.id
-          link_header = response.headers["Link"]
-          link_value = "<#{modify_url_params(request.url, after: next_cursor)}>; rel=\"next\""
-          response.headers["Link"] = link_header ? "#{link_header}, #{link_value}" : link_value
-          response.headers["Pagination-Next"] = next_cursor
-        end
+      if collection_has_more?
+        link_header = response.headers["Link"]
+        link_value = "<#{modify_url_params(request.url, after: last_id_in_collection)}>; rel=\"next\""
+        response.headers["Link"] = link_header ? "#{link_header}, #{link_value}" : link_value
+        response.headers["Pagination-Next"] = last_id_in_collection
       end
     end
 
+    def collection_has_more?
+      return false unless last_id_in_collection
+      remaining_collection = collection.limit(nil).order(id: :asc).where("id > ?", last_id_in_collection)
+      remaining_collection.any?
+    end
+
+    def last_id_in_collection
+      @last_id_in_collection ||= collection&.last&.id
```

Comment:
```suggestion
      @last_id_in_collection ||= collection&.maximum(:id)
````

Small improvement that helps to load less from DB here. It sends the `SELECT MAX("model"."id") ...` query instead of loading the whole model instance and using only the id from there.

Reply:
@Riddlerrr  I think this would result in two queries instead of one. We're always going to have to load the entire collection, so there's really no benefit to only temporarily avoiding it, especially if the cost is an extra query.